osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs

```diff
@@ -142,26 +142,14 @@ private void handleGeneral(string line)
 
                 case @"Mode":
                     beatmap.BeatmapInfo.RulesetID = Parsing.ParseInt(pair.Value);
-
-                    switch (beatmap.BeatmapInfo.RulesetID)
+                    parser = beatmap.BeatmapInfo.RulesetID switch
                     {
-                        case 0:
-                            parser = new Rulesets.Objects.Legacy.Osu.ConvertHitObjectParser(getOffsetTime(), FormatVersion);
-                            break;
-
-                        case 1:
-                            parser = new Rulesets.Objects.Legacy.Taiko.ConvertHitObjectParser(getOffsetTime(), FormatVersion);
-                            break;
-
-                        case 2:
-                            parser = new Rulesets.Objects.Legacy.Catch.ConvertHitObjectParser(getOffsetTime(), FormatVersion);
-                            break;
-
-                        case 3:
-                            parser = new Rulesets.Objects.Legacy.Mania.ConvertHitObjectParser(getOffsetTime(), FormatVersion);
-                            break;
-                    }
-
+                        0 => new Rulesets.Objects.Legacy.Osu.ConvertHitObjectParser(getOffsetTime(), FormatVersion),
+                        1 => new Rulesets.Objects.Legacy.Taiko.ConvertHitObjectParser(getOffsetTime(), FormatVersion),
+                        2 => new Rulesets.Objects.Legacy.Catch.ConvertHitObjectParser(getOffsetTime(), FormatVersion),
+                        3 => new Rulesets.Objects.Legacy.Mania.ConvertHitObjectParser(getOffsetTime(), FormatVersion),
+                        _ => throw new FormatException($"Unknown ruleset ID {beatmap.BeatmapInfo.RulesetID}."),
```

Comment:
A bit of a sneaky semantics change here.

Reply:
Yes, and I think they are necessary. Forgot to write in PR description.